test/k8s/services.go

```diff
@@ -644,6 +644,28 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`,
 			})
 		})
 
+		SkipContextIf(func() bool {
+			return helpers.DoesNotRunWithKubeProxyReplacement() || helpers.RunsOnAKS() || helpers.DoesNotExistNodeWithoutCilium()
+		}, "with L7 policy", func() {
+			var demoPolicyL7 string
+
+			BeforeAll(func() {
+				demoPolicyL7 = helpers.ManifestGet(kubectl.BasePath(), "l7-policy-demo.yaml")
+			})
+
+			AfterAll(func() {
+				kubectl.Delete(demoPolicyL7)
+				// Same reason as in other L7 test above
+				kubectl.CiliumExecMustSucceedOnAll(context.TODO(),
+					"cilium bpf ct flush global", "Unable to flush CT maps")
+			})
+
+			It("Tests NodePort with L7 Policy from outside", func() {
+				applyPolicy(kubectl, demoPolicyL7)
+				testNodePort(kubectl, ni, false, true, 0)
+			})
+		})
```

Comment:
Is this test covered by what's in the CLI connectivity tests or the ci-datapath job?

Reply:
Just checked cilium-cli, there is connectivity test for reaching nodeport from outside, but no L7 policy is applied there to cover #21954: 
https://github.com/cilium/cilium-cli/blob/91329434f04b605e19eb329b8e891f0aa1df49a0/connectivity/tests/service.go#L208-L217
I understand we plan to abandon e2e test suites, I'll open another PR for cilium-cli to cover this scenario after this gets merged.